Updating LaunchDarkly integration docs (#489)
DataDog/integrations-extras: launchdarkly/README.md

# LaunchDarkly Integration

There are currently two integrations with LaunchDarkly:

- [Events Integration](#launchdarkly-events-integration)
- [LD Relay Integration](#ld-relay-integration)

## LaunchDarkly Events Integration

### Overview

Deploying features with LaunchDarkly is great, but sometimes it's difficult to
see the effects of these feature deployments on the customer's services or
systems. Datadog's monitoring and analytics tools for modern DevOps teams allow
their customers to observe the health of their systems over time. The

> LaunchDarkly integration with Datadog exposes the LaunchDarkly events in context

Reviewer: "the" is unnecessary here

with events gathered by the Datadog agents. This makes it easier to reason about
changes to customers' systems as a result of feature deployments that took

> place. For example, if a feature gets deployed that causes a service to slow

Reviewer: "that took place" is unnecessary here.

> down, a DevOps engineer will now be able to see the cause within Datadog.

Reviewer: "will now be able to see" can just be "can see" :)

![LaunchDarkly events in Datadog][7]

### Setup

#### Configuration

> As a prerequisite, this integration will need a [Datadog API

Reviewer: "will need a" -> "requires a"

> key][6]. Note, only Admin users can create an API key. Once you've obtained a Datadog API key, visit the [LaunchDarkly documentation for the Datadog integration][8].

Reviewer: This is a blocker to accomplishing the prerequisite, so I would emphasize it a little bit more than this. Also, starting sentences with "Note," or "Note that" in order to draw reader attention can often have the opposite effect. :) 

Maybe try breaking that sentence out into its own line and rephrasing it as "You must have Administrator privileges in Datadog to create a Datadog API key."

> key][6]. Note, only Admin users can create an API key. Once you've obtained a Datadog API key, visit the [LaunchDarkly documentation for the Datadog integration][8].

Reviewer: "Once" can be a difficult word for non-native English speakers; it's confusing because it can mean either "After you have done the thing" or "Do the thing one time". Let's change this sentence: "After you have the Datadog API key, read about the Datadog integration in [LaunchDarkly's documentation]."

> ### Data Collected

Reviewer: Let's put a stem sentence in to explain what this section does.

Author: How about... "The LaunchDarkly integration sends the following data to Datadog:"

Reviewer: Yeah!

#### Metrics

The LaunchDarkly integration does not include any metrics.

#### Events

> This integration sends LaunchDarkly flag, project, environment events to Datadog. These events can be used to correlate with other metrics available in Datadog.

Reviewer: "These events can be used to..." --> "You can correlate these events with other metrics..."

#### Service Checks

The LaunchDarkly integration does not include any service checks.

### Troubleshooting

Need help? Contact [LaunchDarkly Support][9].

### Further Reading

> Learn more about [LaunchDarkly] and this integration [here][8].

Reviewer: To learn more about LaunchDarkly and the Datadog integration, read [LaunchDarkly's documentation](LINK-TO-DD-INTEGRATION-DOCS).

---

> ## LD Relay Integration

Reviewer: This is a little inconsistent, but even though the feature is called the LD Relay, we refer to it in docs as the "relay proxy." Update the instances in the text below where you refer to it as the "relay" to "relay proxy," please. :)

### Overview

This integration tracks metrics from [LaunchDarkly's relay proxy][1], such as
number of proxied stream connections and proxied API route requests.

> ### Setup

Reviewer: Let's put a stem sentence in to explain what this section does.

#### Configuration

After you've configured the [relay proxy][2], add the following section to your
LaunchDarkly relay instance's `ld-relay.conf` file:

```
[datadog]
enabled=true
statsAddr="YOUR_STATS_ADDRESS"
```

> ### Data Collected

Reviewer: Let's put a stem sentence in to explain what this section does.

#### Metrics

The relay collects metrics on the number of proxied stream connections and

> proxied API route requests. See [relay documentation][3] for more information.

Reviewer: To learn more, read the [relay proxy documentation].

#### Events

The LaunchDarkly integration does not include any events.

#### Service Checks

The LaunchDarkly integration does not include any service checks.

### Troubleshooting

Need help? Contact Datadog [Support][4].

### Further Reading

Learn more about infrastructure monitoring and all our integrations on [our

> blog][5].

Reviewer: Can we put a link to the LD docs here too? We have a bunch of docs about all our integrations.

# LaunchDarkly Integration

> ### Overview

Reviewer 2: ```suggestion
## Overview
```

Let's start with an H2 instead of an H3 to benefit from the automated ToC from Datadog Docs.

Deploying features with LaunchDarkly is great, but it can be difficult to see
the effects of these deployments on users' services or systems. Datadog's
monitoring and analytics tools for modern DevOps teams allow their customers to
observe the health of their systems over time. The LaunchDarkly integration with
Datadog exposes LaunchDarkly events in context with events gathered by the
Datadog agents. This makes it easier to reason about changes to customers'
systems as a result of feature deployments. For example, if a feature gets
deployed that causes a service to slow down, a DevOps engineer can see the cause

> within Datadog.

Reviewer 2: ```suggestion
The Datadog-LaunchDarkly integration allows you to see the effects of your LaunchDarkly feature deployments on your customer's services or systems. For instance, if a deployed feature causes a service to slow down, you would be able to see the cause within Datadog.
This integration displays LaunchDarkly events in the [Datadog Event Stream](https://app.datadoghq.com/event/stream) and collect metrics from [LaunchDarkly's relay proxy](https://docs.launchdarkly.com/docs/the-relay-proxy), such as the number of proxied stream connections and proxied API route requests.
```

There is no need to re-explain what LaunchDarkly and Datadog are and do here, only what's the end-goal of this integration.
From a style perspective there is no need to break lines to respect a max number of characters. It's a doc not a piece of code :)

![LaunchDarkly events in Datadog][2]

> ### Setup

Reviewer 2: ```suggestion
## Setup
```

> #### Configuration

Reviewer 2: ```suggestion
```

Since there is only one part to setup this integration let's remove un-necessary header.

> As a prerequisite, this integration will need a [Datadog API key][1]. Note, only

Reviewer: Let's clarify here that you need Datadog admin permissions, and rephrase this slightly:

"This integration requires a Datadog API key. Only Datadog admins can create an API key in Datadog."

create an API key. Once you've obtained a Datadog API key, visit the
[LaunchDarkly documentation for the Datadog integration][3] for further

> instructions on how to register the API key in LaunchDarkly.

Reviewer 2: ```suggestion
Visit the [LaunchDarkly documentation for the Datadog integration][3] to learn how to setup the Datadog-LaunchDarky integration.

**Note**: In order to set it up, you need a valid [Datadog API key][1].
```

All instructions are already outlined in the LaunchDarkly doc, no need to duplicate them here :)

> ### Data Collected

Reviewer 2: ```suggestion
## Data Collected
```

> The LaunchDarkly integration sends the following data to Datadog:

Reviewer 2: Right now it only sends events though. Do you know if we could get a list of all metrics collected by this integration in oder to display them here ? And say something along the lines of

`Find below a list of metrics that are collected after [enabling XXX]()`

Author: This integration doesn't send any metrics to DD. I wasn't sure if it was a requirement to have a Metrics section. I just copied what was there from the prior readme.

> #### Metrics

Reviewer 2: ```suggestion
### Metrics
```

The LaunchDarkly integration does not include any metrics.

> #### Events

Reviewer 2: ```suggestion
### Events
```

This integration sends LaunchDarkly flag, project, environment events to
Datadog. These events can be used to correlate with other metrics available in

> Datadog.

Reviewer 2: ```suggestion
The LaunchDarkly integration sends the flag, project, and environment LaunchDarkly events to Datadog.
```

Slight wording update to match the one from Metrics/SC data collection.

> #### Service Checks

Reviewer 2: ```suggestion
### Service Checks
```

The LaunchDarkly integration does not include any service checks.

> ### Troubleshooting

Reviewer 2: ```suggestion
## Troubleshooting
```

Need help? Contact [LaunchDarkly Support][4].

> ### Further Reading

Reviewer 2: ```suggestion
## Further Reading
```
